# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1369937999 14400
#      Thu May 30 14:19:59 2013 -0400
# Node ID 736dca8371eee979e810c40865da592af24c8c11
# Parent  122d3f62e179ba044d47e58510905719220d8706
include more signals in SIG function (bug #39089)

* siglist.c (init_signals): Correct SIGPHONE case to use SIGPHONE
index rather than SIGWIND.  Add SIGSTKFLT and SIGUNUSED cases.
* sighandlers.cc (make_sig_struct): Move SIGIO to alphabetical
order.  Assign SIGKILL, SIGSTKFLT, and SIGUNUSED cases.

diff --git a/libinterp/interp-core/siglist.c b/libinterp/interp-core/siglist.c
--- a/libinterp/interp-core/siglist.c
+++ b/libinterp/interp-core/siglist.c
@@ -117,17 +117,17 @@ init_signals (void)
 # endif
 # ifdef SIGLWP
       sys_siglist[SIGLWP] = "SIGLWP";
 # endif
 # ifdef SIGMSG
       sys_siglist[SIGMSG] = "Monitor mode data available";
 # endif
 # ifdef SIGPHONE
-      sys_siglist[SIGWIND] = "SIGPHONE";
+      sys_siglist[SIGPHONE] = "SIGPHONE";
 # endif
 # ifdef SIGPIPE
       sys_siglist[SIGPIPE] = "Broken pipe";
 # endif
 # ifdef SIGPOLL
       sys_siglist[SIGPOLL] = "Pollable event occurred";
 # endif
 # ifdef SIGPROF
@@ -149,16 +149,19 @@ init_signals (void)
       sys_siglist[SIGSAK] = "Secure attention";
 # endif
 # ifdef SIGSEGV
       sys_siglist[SIGSEGV] = "Segmentation violation";
 # endif
 # ifdef SIGSOUND
       sys_siglist[SIGSOUND] = "Sound completed";
 # endif
+# ifdef SIGSTKFLT
+      sys_siglist[SIGSTKFLT] = "Stack fault";
+# endif
 # ifdef SIGSTOP
       sys_siglist[SIGSTOP] = "Stopped (signal)";
 # endif
 # ifdef SIGSTP
       sys_siglist[SIGSTP] = "Stopped (user)";
 # endif
 # ifdef SIGSYS
       sys_siglist[SIGSYS] = "Bad argument to system call";
@@ -176,16 +179,19 @@ init_signals (void)
       sys_siglist[SIGTSTP] = "Stopped (user)";
 # endif
 # ifdef SIGTTIN
       sys_siglist[SIGTTIN] = "Stopped (tty input)";
 # endif
 # ifdef SIGTTOU
       sys_siglist[SIGTTOU] = "Stopped (tty output)";
 # endif
+# ifdef SIGUNUSED
+      sys_siglist[SIGUNUSED] = "SIGUNUSED";
+# endif
 # ifdef SIGURG
       sys_siglist[SIGURG] = "Urgent I/O condition";
 # endif
 # ifdef SIGUSR1
       sys_siglist[SIGUSR1] = "User defined signal 1";
 # endif
 # ifdef SIGUSR2
       sys_siglist[SIGUSR2] = "User defined signal 2";
diff --git a/libinterp/interpfcn/sighandlers.cc b/libinterp/interpfcn/sighandlers.cc
--- a/libinterp/interpfcn/sighandlers.cc
+++ b/libinterp/interpfcn/sighandlers.cc
@@ -625,20 +625,28 @@ make_sig_struct (void)
 #ifdef SIGINFO
   m.assign ("INFO", SIGINFO);
 #endif
 
 #ifdef SIGINT
   m.assign ("INT", SIGINT);
 #endif
 
+#ifdef SIGIO
+  m.assign ("IO", SIGIO);
+#endif
+
 #ifdef SIGIOT
   m.assign ("IOT", SIGIOT);
 #endif
 
+#ifdef SIGKILL
+  m.assign ("KILL", SIGKILL);
+#endif
+
 #ifdef SIGLOST
   m.assign ("LOST", SIGLOST);
 #endif
 
 #ifdef SIGPIPE
   m.assign ("PIPE", SIGPIPE);
 #endif
 
@@ -657,16 +665,20 @@ make_sig_struct (void)
 #ifdef SIGQUIT
   m.assign ("QUIT", SIGQUIT);
 #endif
 
 #ifdef SIGSEGV
   m.assign ("SEGV", SIGSEGV);
 #endif
 
+#ifdef SIGSTKFLT
+  m.assign ("STKFLT", SIGSTKFLT);
+#endif
+
 #ifdef SIGSTOP
   m.assign ("STOP", SIGSTOP);
 #endif
 
 #ifdef SIGSYS
   m.assign ("SYS", SIGSYS);
 #endif
 
@@ -685,36 +697,36 @@ make_sig_struct (void)
 #ifdef SIGTTIN
   m.assign ("TTIN", SIGTTIN);
 #endif
 
 #ifdef SIGTTOU
   m.assign ("TTOU", SIGTTOU);
 #endif
 
+#ifdef SIGUNUSED
+  m.assign ("UNUSED", SIGUNUSED);
+#endif
+
 #ifdef SIGURG
   m.assign ("URG", SIGURG);
 #endif
 
 #ifdef SIGUSR1
   m.assign ("USR1", SIGUSR1);
 #endif
 
 #ifdef SIGUSR2
   m.assign ("USR2", SIGUSR2);
 #endif
 
 #ifdef SIGVTALRM
   m.assign ("VTALRM", SIGVTALRM);
 #endif
 
-#ifdef SIGIO
-  m.assign ("IO", SIGIO);
-#endif
-
 #ifdef SIGWINCH
   m.assign ("WINCH", SIGWINCH);
 #endif
 
 #ifdef SIGXCPU
   m.assign ("XCPU", SIGXCPU);
 #endif
 
